### DISCUSSION TITLE
`IUpdatableByPlayfield.Update` gets called multiple times per frame by nested playfields

### DISCUSSION THREAD

**Discussion by hlysine** [ppy/osu#17064] [Ideas] [answered]
Re: https://discord.com/channels/188630481301012481/188630652340404224/948794610271014923

Currently, all playfields, nested or not, find `IUpdatableByPlayfield` mods and update them every frame. As a result, rulesets with nested playfields update mods multiple times per frame, which may not be expected. In fact, most of the existing mods expect a single update per frame, so we need a way for mods to get that.

Some possible solutions suggested by @peppy:

1. Add a second method in `IUpdatableByPlayfield`:
```csharp
    public interface IUpdatableByPlayfield : IApplicableMod
    {
        void Update(Playfield playfield);

        void UpdateNested(Playfield nestedPlayfield) { }
    }
```

2. Add a new interface `IUpdatableByPlayer` to update mods once per frame.

**Comment by peppy** [comment, marked as answer]
Of note, *all* existing usages of this expect one update per game frame, and as such this is pretty damn broken.

**Reply by hlysine** [reply]
Can't we just allow only the main playfield to update mods then? Nested playfields can be accessed through `Playfield.NestedPlayfields` if mods really need to do something in nested playfields.

**Reply by peppy** [reply]
That is a potential solution, alongside documentation changes specifying this.

**Reply by hlysine** [reply]
PR created: #17069
